File: javascript/src/math/gcd.js
/**
 * ユークリッドの互除法を用いて最大公約数を計算します。
 * 
 * @param {number} a 整数
 * @param {number} b 整数
 * @returns {number} 最大公約数
 */
function gcd(a, b) {
    a = Math.abs(a);
    b = Math.abs(b);
    while (b) {
        [a, b] = [b, a % b];
    }
    return a;
}

module.exports = { gcd };

File: python/src/basic_math/gcd.py
def gcd(a: int, b: int) -> int:
    """
    ユークリッドの互除法を用いて最大公約数を計算します。
    
    Args:
        a (int): 整数
        b (int): 整数
        
    Returns:
        int: 最大公約数
    """
    while b:
        a, b = b, a % b
    return abs(a)
